node/network/approval-distribution/src/lib.rs

```diff
@@ -478,17 +735,21 @@ impl State {
 		}
 	}
 
+	// handle a peer view change: requires that the peer is already connected
+	// and has an entry in the `PeerData` struct.
 	async fn handle_peer_view_change(
 		&mut self,
 		ctx: &mut (impl SubsystemContext<Message = ApprovalDistributionMessage>
 		          + overseer::SubsystemContext<Message = ApprovalDistributionMessage>),
 		metrics: &Metrics,
 		peer_id: PeerId,
 		view: View,
+		rng: &mut (impl CryptoRng + Rng),
 	) {
 		gum::trace!(target: LOG_TARGET, ?view, "Peer view change");
 		let finalized_number = view.finalized_number;
-		let old_view = self.peer_views.insert(peer_id.clone(), view.clone());
+		let old_view =
+			self.peer_views.get_mut(&peer_id).map(|d| std::mem::replace(d, view.clone()));
```

Comment:
Isn't this identical to the previous impl?

Reply:
Not quite. i rewrote some code so that `peer_views` always has an entry for any connected peers. I'm explicitly not `insert`ing.

Reply:
Ah, yes